### PR TITLE
feat: Add content sanitization for non-text MCP data types

### PR DESCRIPTION
**Pull request by naji247**
## Summary
- Add two-layer content sanitization to detect and redact non-text data from events before they reach the MCPCat API and telemetry exporters
- **Layer 1 (Response):** Replace non-text content blocks (image, audio, blob resources, unknown types) with informative redaction messages while passing through text and resource_link blocks
- **Layer 2 (Parameters):** Recursively scan parameters for large base64-encoded strings (>=10KB size gate) and redact them

## Test plan
- [x] Run full test suite: `pnpm test` (321 passing, 0 regressions)
- [x] Run sanitization tests: `pnpm test src/tests/sanitization.test.ts` (19 tests covering response blocks, parameter scanning, structuredContent, boundary conditions, and immutability)
- [ ] Verify an MCP server returning `ImageContent` in a tool response logs the redaction message in `~/mcpcat.log` instead of base64 data

### CODE REVIEW

File: src/modules/eventQueue.ts [hunk 1]
```diff
@@ -50,27 +51,27 @@ class EventQueue {
...
-      if (event) {
-        event.id = event.id || (await KSUID.withPrefix("evt").random());
-        this.activeRequests++;
-        this.sendEvent(event as Event).finally(() => {
-          this.activeRequests--;
-          // Try to process more events
-          this.process();
-        });
-      }
+      Object.assign(event, sanitizeEvent(event));
```
Comment (kashishhora) [inline src/modules/eventQueue.ts]: Should put this call inside a try/catch just in case sanitization fails so that it doesn't block the queue.

Comment (kashishhora) [reply src/modules/eventQueue.ts]: Honestly, at this point might even consider wrapping the process() call itself in a try/catch and then having this.processing = false in the finally block? That way we also capture/log any issues in processing and don't block the queue for those either? But that's more optional.

File: src/modules/sanitization.ts [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+
+  if (Array.isArray(obj)) {
+    return obj.map(sanitizeParameters);
+  }
+
+  if (obj instanceof Date) {
+    return obj;
+  }
+
+  if (typeof obj === "object") {
```
Comment (kashishhora) [inline src/modules/sanitization.ts:129]: Nit: assumption here is that all objects will be plain objects, so this might cause issues with e.g. Buffer objects but not sure how important that is to factor in so I'll leave it up to you.

**Review by kashishhora — CHANGES_REQUESTED** [review]

**Review by kashishhora — APPROVED** [review]
